src/levels/procedural_test/levelgen.gd

```diff
@@ -3,63 +3,87 @@ extends Node2D
 export var level_size:=Vector2(3,35)
 export var unit_size = 32
 export var chunk_size = 20
-export var budget = 100
-export var grace_period = 3
 
 var chunks:= []
 
+#NEW
+var last_chunk
+var last_position
+var difficulty_start = 3
+var difficulty_modifier = 1.1
+var budget:int
```

Comment:
If you're gonna set types, might as well set them everywhere for clarity. Not a dealbreaker though. ✌️